### PR TITLE
Added Accordion paragraph

### PR DESCRIPTION
**Pull request by gosia-mlynarczyk**
Added 'Accordion' paragraph (and its child 'Accordion pane' paragraph).

Since this is my first pull request I'd appreciate a thorough review and feedback 🙂

### CODE REVIEW

**Review by stephen-cox — CHANGES_REQUESTED** [review]
Hi @gosia-mlynarczyk, the accordion functionality seems to work well.

There is a small bit of configuration missing - you need to allow the Accordion paragraph type to be used as page content (/admin/structure/types/manage/localgov_campaigns_overview/fields/node.localgov_campaigns_overview.localgov_campaigns_content)

We decided on some [naming conventions](https://github.com/localgovdrupal/localgov/wiki/Naming-conventions) to drop the `field_` prefix and use `localgov_` instead, so the newly defined fields should ideally be renamed.

To rename the fields you can do something like the following from the `config/install` directory (which assumes you have sed and rename installed):

```
sed -i 's/field_name/localgov_name/g' *
rename 's/field_name/localgov_name/g' *
```

There is also a coding standards issue which means the automated tests are failing. Looks like the `_create_heading` function should return `null|array`:

```
FILE: ...calgov_campaigns_paragraphs/localgov_campaigns_paragraphs.module
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 112 | ERROR | Function return type is not void, but function is
     |       | returning void here
----------------------------------------------------------------------
```

Then only thing I'm unsure of code-wise is why you're using the `_create_heading` function to create a heading rather than do the whole thing in the Twig template. It's not a major thing, just wondering why you chose to do it that way?

**Comment by gosia-mlynarczyk** [comment]
Thanks @stephen-cox , that's very helpful!

Yeah, I wasn't sure about naming convention for fields, as some of them are prefixed with `field_` and some with `localgov_`. I'll change it.

Re: headings & heading levels, there's actually a module (https://www.drupal.org/project/heading) which provides a 'heading' field type. So rather than adding a 'title' and 'heading level' fields to paragraphs we would only need to add 1 'heading' field. Do you think it would be better to use that module instead?

**Comment by stephen-cox** [comment]
> Re: headings & heading levels, there's actually a module (https://www.drupal.org/project/heading) which provides a 'heading' field type. So rather than adding a 'title' and 'heading level' fields to paragraphs we would only need to add 1 'heading' field. Do you think it would be better to use that module instead?

It seems a simple thing to have a two fields, title and heading level, but if the module provides extra benefits then it's fine to use it. It does provide an extra dependency that we have to manage, so we need to weigh the benefits against the extra work with upgrades and other potentially breaking changes introduced by contrib modules.

I'm happy with the field structure as it stands. My query was really over what the benefits are of using the `_create_heading` function to process the title and heading level were over doing all the display stuff in the `paragraph--localgov-accordion-pane.html.twig` template file.

**Comment by gosia-mlynarczyk** [comment]
Ok, I'll leave those 2 separate fields then rather than using a contrib module.

The reason I created a custom `_create_heading` function is that there will be more paragraphs using title / heading level combo. But I could just create that heading element in each paragraph's template.

**Comment by stephen-cox** [comment]
> The reason I created a custom `_create_heading` function is that there will be more paragraphs using title / heading level combo. But I could just create that heading element in each paragraph's template.

That's fine. Just wondering why you took that approach.

If you rename the fields and fix the coding standards error I'm happy to approve this merge request.

File: modules/localgov_campaigns_paragraphs/localgov_campaigns_paragraphs.module [hunk 1]
```diff
@@ -76,4 +84,42 @@ function localgov_campaigns_paragraphs_preprocess_paragraph(&$variables) {
...
+ *
+ * @param string $heading_text
+ *   Heading text.
+ * @param string $heading_level
+ *   Heading level.
+ *
+ * @return void|array
+ *   Heading render element.
+ */
+function _create_heading(string $heading_text, string $heading_level) {
```
Comment (Adnan-cds) [inline modules/localgov_campaigns_paragraphs/localgov_campaigns_paragraphs.module:108]: Good morning Gosia :)  If you don't mind, can we have this function be called _localgov_campaigns_paragraphs_create_heading().

Comment (gosia-mlynarczyk) [reply modules/localgov_campaigns_paragraphs/localgov_campaigns_paragraphs.module]: Done! :)

**Pull request merged**